paimon-core/src/main/java/org/apache/paimon/operation/AppendOnlyFileStoreWrite.java

```diff
@@ -147,7 +147,7 @@ public AppendOnlyCompactManager.CompactRewriter compactRewriter(
             if (toCompact.isEmpty()) {
                 return Collections.emptyList();
             }
-            RowDataRollingFileWriter rewriter =
+            try (RowDataRollingFileWriter rewriter =
```

Comment:
Can you use try finally to avoid closing twice?

Reply:
Sry forget to remove the close inside.

Reply:
But `rewriter.result()` requires that the `writer` has been closed.